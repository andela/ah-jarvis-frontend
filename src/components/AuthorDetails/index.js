import StarRatingComponent from 'react-star-rating-component';
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import capitalize from '../../utils/capitalize';
import getCurrentUser from '../../utils/auth';
import ROUTES from '../../utils/routes';

const AuthorDetails = ({
  user, date, small, averageRate, onStarClick,
}) => (
  <div className="row">
    <div className={`author ${!small && 'col m9'}`}>
      <Link to="/">
        <div className="author__avatar">
          <img src={user.image} alt="" className={`responsive-img circle ${small ? 'avatar--small' : 'avatar--small'}`} />
        </div>
      </Link>
      <div className={`author_details ${small && 'author--small'}`}>
        <div className="author__info">
          <div className="author-name">
            <Link className="text--small" to={`${ROUTES.profile}/${user.username}`}>{capitalize(user.username)}</Link>
          </div>
          {!small && <a className="btn-flat btn-flat--primary" href="#!">Follow</a>}
        </div>
        <div className="article__highlights">
          {new Date(date).toLocaleDateString(undefined, {
            month: 'short',
            day: 'numeric',
            year: 'numeric',
          })}
          {small ? (
            <React.Fragment>
              <span className="p-r--5 p-l--5">.</span>
              13 min read
            </React.Fragment>
          ) : (
            <React.Fragment>
              <span className="p-r--5 p-l--5">.</span>
              13 min read
            </React.Fragment>
          ) }
          <span className="p-r--10 p-l--10">.</span>

          {!small && (
          <React.Fragment>
            <span className="p-r--10 p-l--10">
              <StarRatingComponent
                name="rate1"
                starCount={5}
                value={averageRate}
                starColor="#03a87c"
                emptyStarColor="#ccc"
                onStarClick={onStarClick}
              />
            </span>
            <p>{ averageRate }</p>
          </React.Fragment>
          )}

        </div>
      </div>
    </div>

    <div className="col m3">
      <div className="m-t--15">
        {!small
        && getCurrentUser() && user.username === getCurrentUser().username && (
        <button type="button" className="btn-flat btn-flat--primary">
          Edit Article
        </button>
        )
      }
      </div>
    </div>
  </div>
);

export default AuthorDetails;

AuthorDetails.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }).isRequired,
  small: PropTypes.bool,
  date: PropTypes.string.isRequired,
  onStarClick: PropTypes.func,
  averageRate: PropTypes.number,
};
