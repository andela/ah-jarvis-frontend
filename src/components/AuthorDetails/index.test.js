import { shallow } from 'enzyme';
import React from 'react';
import AuthorDetails from '.';

describe('<AuthorDetails />', () => {
  it('renders  <AuthorDetails /> components', () => {
    const snap = shallow(<AuthorDetails user={{ username: 'test user' }} date="" />);
    expect(snap).toMatchSnapshot();
  });
});
