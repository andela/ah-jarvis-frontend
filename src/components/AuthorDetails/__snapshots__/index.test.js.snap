// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorDetails /> renders  <AuthorDetails /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthorDetails
    date=""
    small={true}
    user={
        Object {
            "username": "test user",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="author col m10"
>
          <Link
                    replace={false}
                    to="/"
          >
                    <div
                              className="author__avatar"
                    >
                              <img
                                        alt="test user"
                                        className="responsive-img circle avatar--small"
                                        src={undefined}
                              />
                    </div>
          </Link>
          <div
                    className="author_details author--small"
          >
                    <div
                              className="author__info"
                    >
                              <div
                                        className="author-name"
                              >
                                        <Link
                                                  className="text--small"
                                                  replace={false}
                                                  to="/profile/test user"
                                        >
                                                  Test User
                                        </Link>
                              </div>
                    </div>
                    <div
                              className="article__highlights"
                    >
                              Invalid Date
                              <Unknown>
                                        <span
                                                  className="p-r--5 p-l--5"
                                        >
                                                  .
                                        </span>
                                        undefined min read
                              </Unknown>
                    </div>
          </div>
</div>,
        <div
          className="col m2 right"
>
          <div
                    className="m-t--15"
          >
                    
          </div>
</div>,
      ],
      "className": "row author__panel p-t--20",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
>
              <div
                            className="author__avatar"
              >
                            <img
                                          alt="test user"
                                          className="responsive-img circle avatar--small"
                                          src={undefined}
                            />
              </div>
</Link>,
            <div
              className="author_details author--small"
>
              <div
                            className="author__info"
              >
                            <div
                                          className="author-name"
                            >
                                          <Link
                                                        className="text--small"
                                                        replace={false}
                                                        to="/profile/test user"
                                          >
                                                        Test User
                                          </Link>
                            </div>
              </div>
              <div
                            className="article__highlights"
              >
                            Invalid Date
                            <Unknown>
                                          <span
                                                        className="p-r--5 p-l--5"
                                          >
                                                        .
                                          </span>
                                          undefined min read
                            </Unknown>
              </div>
</div>,
          ],
          "className": "author col m10",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="author__avatar"
>
                <img
                                alt="test user"
                                className="responsive-img circle avatar--small"
                                src={undefined}
                />
</div>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="test user"
                  className="responsive-img circle avatar--small"
                  src={undefined}
/>,
                "className": "author__avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "test user",
                  "className": "responsive-img circle avatar--small",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="author__info"
>
                  <div
                                    className="author-name"
                  >
                                    <Link
                                                      className="text--small"
                                                      replace={false}
                                                      to="/profile/test user"
                                    >
                                                      Test User
                                    </Link>
                  </div>
</div>,
                <div
                  className="article__highlights"
>
                  Invalid Date
                  <Unknown>
                                    <span
                                                      className="p-r--5 p-l--5"
                                    >
                                                      .
                                    </span>
                                    undefined min read
                  </Unknown>
</div>,
              ],
              "className": "author_details author--small",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="author-name"
>
                      <Link
                                            className="text--small"
                                            replace={false}
                                            to="/profile/test user"
                      >
                                            Test User
                      </Link>
</div>,
                    false,
                  ],
                  "className": "author__info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="text--small"
                        replace={false}
                        to="/profile/test user"
>
                        Test User
</Link>,
                      "className": "author-name",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Test User",
                        "className": "text--small",
                        "replace": false,
                        "to": "/profile/test user",
                      },
                      "ref": null,
                      "rendered": "Test User",
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  false,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Invalid Date",
                    <Unknown>
                      <span
                                            className="p-r--5 p-l--5"
                      >
                                            .
                      </span>
                      undefined min read
</Unknown>,
                    false,
                  ],
                  "className": "article__highlights",
                },
                "ref": null,
                "rendered": Array [
                  "Invalid Date",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <span
                          className="p-r--5 p-l--5"
>
                          .
</span>,
                        "undefined min read",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": ".",
                          "className": "p-r--5 p-l--5",
                        },
                        "ref": null,
                        "rendered": ".",
                        "type": "span",
                      },
                      "undefined min read",
                    ],
                    "type": Symbol(react.fragment),
                  },
                  false,
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="m-t--15"
>
            
</div>,
          "className": "col m2 right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              null,
            ],
            "className": "m-t--15",
          },
          "ref": null,
          "rendered": Array [
            null,
            null,
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="author col m10"
>
            <Link
                        replace={false}
                        to="/"
            >
                        <div
                                    className="author__avatar"
                        >
                                    <img
                                                alt="test user"
                                                className="responsive-img circle avatar--small"
                                                src={undefined}
                                    />
                        </div>
            </Link>
            <div
                        className="author_details author--small"
            >
                        <div
                                    className="author__info"
                        >
                                    <div
                                                className="author-name"
                                    >
                                                <Link
                                                            className="text--small"
                                                            replace={false}
                                                            to="/profile/test user"
                                                >
                                                            Test User
                                                </Link>
                                    </div>
                        </div>
                        <div
                                    className="article__highlights"
                        >
                                    Invalid Date
                                    <Unknown>
                                                <span
                                                            className="p-r--5 p-l--5"
                                                >
                                                            .
                                                </span>
                                                undefined min read
                                    </Unknown>
                        </div>
            </div>
</div>,
          <div
            className="col m2 right"
>
            <div
                        className="m-t--15"
            >
                        
            </div>
</div>,
        ],
        "className": "row author__panel p-t--20",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
>
                <div
                                className="author__avatar"
                >
                                <img
                                                alt="test user"
                                                className="responsive-img circle avatar--small"
                                                src={undefined}
                                />
                </div>
</Link>,
              <div
                className="author_details author--small"
>
                <div
                                className="author__info"
                >
                                <div
                                                className="author-name"
                                >
                                                <Link
                                                                className="text--small"
                                                                replace={false}
                                                                to="/profile/test user"
                                                >
                                                                Test User
                                                </Link>
                                </div>
                </div>
                <div
                                className="article__highlights"
                >
                                Invalid Date
                                <Unknown>
                                                <span
                                                                className="p-r--5 p-l--5"
                                                >
                                                                .
                                                </span>
                                                undefined min read
                                </Unknown>
                </div>
</div>,
            ],
            "className": "author col m10",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="author__avatar"
>
                  <img
                                    alt="test user"
                                    className="responsive-img circle avatar--small"
                                    src={undefined}
                  />
</div>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="test user"
                    className="responsive-img circle avatar--small"
                    src={undefined}
/>,
                  "className": "author__avatar",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "test user",
                    "className": "responsive-img circle avatar--small",
                    "src": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="author__info"
>
                    <div
                                        className="author-name"
                    >
                                        <Link
                                                            className="text--small"
                                                            replace={false}
                                                            to="/profile/test user"
                                        >
                                                            Test User
                                        </Link>
                    </div>
</div>,
                  <div
                    className="article__highlights"
>
                    Invalid Date
                    <Unknown>
                                        <span
                                                            className="p-r--5 p-l--5"
                                        >
                                                            .
                                        </span>
                                        undefined min read
                    </Unknown>
</div>,
                ],
                "className": "author_details author--small",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="author-name"
>
                        <Link
                                                className="text--small"
                                                replace={false}
                                                to="/profile/test user"
                        >
                                                Test User
                        </Link>
</div>,
                      false,
                    ],
                    "className": "author__info",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          className="text--small"
                          replace={false}
                          to="/profile/test user"
>
                          Test User
</Link>,
                        "className": "author-name",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Test User",
                          "className": "text--small",
                          "replace": false,
                          "to": "/profile/test user",
                        },
                        "ref": null,
                        "rendered": "Test User",
                        "type": [Function],
                      },
                      "type": "div",
                    },
                    false,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Invalid Date",
                      <Unknown>
                        <span
                                                className="p-r--5 p-l--5"
                        >
                                                .
                        </span>
                        undefined min read
</Unknown>,
                      false,
                    ],
                    "className": "article__highlights",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Invalid Date",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          <span
                            className="p-r--5 p-l--5"
>
                            .
</span>,
                          "undefined min read",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": ".",
                            "className": "p-r--5 p-l--5",
                          },
                          "ref": null,
                          "rendered": ".",
                          "type": "span",
                        },
                        "undefined min read",
                      ],
                      "type": Symbol(react.fragment),
                    },
                    false,
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="m-t--15"
>
              
</div>,
            "className": "col m2 right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                null,
              ],
              "className": "m-t--15",
            },
            "ref": null,
            "rendered": Array [
              null,
              null,
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
